CLI compile errors on Ubuntu 24.04

## How to reproduce

On Ubuntu 24.04:

```bash
sudo apt install cmake build-essential clang
sudo apt install libboost-all-dev libfreetype-dev 

git clone --recursive git@github.com:maplibre/font-maker.git
cd font-maker/

cmake .
make
```

Dependency versions: Clang 18.1.3, Boost 1.83.0, Freetype 2.13.2.


Got type narrowing errors on `sdf-glyph-foundry`, such as:

```
~/font-maker/vendor/sdf-glyph-foundry/include/mapbox/glyph_foundry_impl.hpp:186:11: error: non-constant-expression cannot be narrowed from type 'double' to 'float' in initializer list [-Wc++11-narrowing-const-reference]
  186 |           v.get<0>() + t * (w.get<0>() - v.get<0>()),
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Could fix it by adding the line `set(CMAKE_CXX_FLAGS "-Wno-c++11-narrowing")` in `CMakeLists.txt`.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,6 +4,7 @@ set(CMAKE_CXX_COMPILER "/usr/bin/clang++")
 project(font-maker)
 set(CMAKE_CXX_FLAGS_RELEASE "-O3")
 set(CMAKE_CXX_FLAGS_DEBUG "-DDEBUG -g")
+set(CMAKE_CXX_FLAGS "-Wno-c++11-narrowing")
 
 find_package(Boost 1.73 REQUIRED)
 find_package(Freetype REQUIRED)

